master/internal/proxy/proxy_intg_test.go

```diff
@@ -0,0 +1,100 @@
+package proxy
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	proxyAuth = func(c echo.Context) (done bool, err error) {
+		return true, nil
+	}
+	serviceIDs = []string{"a", "b", "c"}
+	u          = url.URL{}
+)
+
+func register(t *testing.T, prTCP bool, unauth bool) {
+	for _, id := range serviceIDs {
+		DefaultProxy.Register(id, &u, prTCP, unauth)
+		if DefaultProxy.GetService(id) == nil {
+			t.Errorf("failed to find registered service %s", id)
+		}
+	}
+	if len(DefaultProxy.Summary()) != len(serviceIDs) {
+		t.Errorf("failed to register all services")
+	}
+}
+
+func unregister(t *testing.T) {
+	for _, id := range serviceIDs {
+		if DefaultProxy.GetService(id) == nil {
+			t.Errorf("failed to find registered service %s", id)
+		}
+		DefaultProxy.Unregister(id)
+		if DefaultProxy.GetService(id) != nil {
+			t.Errorf("failed to unregister service %s", id)
+		}
+	}
+}
+
+func TestProxyLifecycle(t *testing.T) {
```

Comment:
can we add a todo + ticket for testing the handler?

Reply:
made it a ticket -- https://hpe-aiatscale.atlassian.net/browse/DET-9555